### PR TITLE
Add camel-atom integration-test

### LINKED ISSUE #410: Add camel-atom integration test
camel-atom integration test

### PR DESCRIPTION
**Pull request by f2par0**
Fixes #410 
Add camel-atom integration test based on apache camel test

### CODE REVIEW

File: tests/features/camel-atom/src/main/java/org/apache/karaf/camel/test/CamelAtomRouteSupplier.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        return builder ->
+                builder.fromF("atom://file:%s/test-classes/data/feed.atom?delay=500",System.getProperty("project.target"))
+                        .log("received message ${body}");
+    }
+
+    protected boolean producerEnabled() {
+        return false;
+    }
+
+
```
Comment (essobedo) [inline tests/features/camel-atom/src/main/java/org/apache/karaf/camel/test/CamelAtomRouteSupplier.java:44]:
```suggestion
```

File: tests/features/camel-atom/src/test/java/org/apache/karaf/camel/itest/CamelAtomITest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                "a great presentation on REST, JAX-WS and JSR 311",
+                "my slides on ActiveMQ and Camel from last weeks Dublin Conference",
+                "webcast today on Apache ActiveMQ",
+                "Feedback on my Camel talk at the IJTC conference",
+                "More thoughts on RESTful Message Queues",
+                "ActiveMQ webinar archive available" };
+        int counter = 0;
+        for (Exchange exchange : list) {
+            Item entry = exchange.getIn().getBody(Item.class);
+            assertNotNull( "No entry found for exchange: " + exchange,entry);
```
Comment (essobedo) [inline tests/features/camel-atom/src/test/java/org/apache/karaf/camel/itest/CamelAtomITest.java:53]:
```suggestion
            assertNotNull("No entry found for exchange: " + exchange, entry);
```

File: tests/features/camel-atom/src/test/java/org/apache/karaf/camel/itest/CamelAtomITest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import org.apache.karaf.camel.itests.CamelKarafTestHint;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+import com.apptasticsoftware.rssreader.Item;
+
+@CamelKarafTestHint
```
Comment (essobedo) [inline tests/features/camel-atom/src/test/java/org/apache/karaf/camel/itest/CamelAtomITest.java:33]: Is it really needed?

File: tests/features/camel-atom/src/test/java/org/apache/karaf/camel/itest/CamelAtomITest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+            counter++;
+        }
+    }
+
+    @Test
+    public void testResultMock() throws Exception {
+        assertMockEndpointsSatisfied();
+    }
+
```
Comment (essobedo) [inline tests/features/camel-atom/src/test/java/org/apache/karaf/camel/itest/CamelAtomITest.java:67]:
```suggestion
```

File: tests/features/camel-atom/src/main/java/org/apache/karaf/camel/test/CamelAtomRouteSupplier.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        name = "karaf-camel-atom-test",
+        immediate = true,
+        service = CamelAtomRouteSupplier.class
+)
+public class CamelAtomRouteSupplier extends AbstractCamelSingleFeatureResultMockBasedRouteSupplier {
+
+    @Override
+    protected Function<RouteBuilder, RouteDefinition> consumerRoute() {
+        return builder ->
+                builder.fromF("atom://file:%s/test-classes/data/feed.atom?delay=500",System.getProperty("project.target"))
```
Comment (essobedo) [inline tests/features/camel-atom/src/main/java/org/apache/karaf/camel/test/CamelAtomRouteSupplier.java:36]:
```suggestion
                builder.fromF("atom://file:%s/test-classes/data/feed.atom?delay=500", System.getProperty("project.target"))
```

File: tests/features/camel-atom/src/main/java/org/apache/karaf/camel/test/CamelAtomRouteSupplier.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+package org.apache.karaf.camel.test;
+
+import java.util.function.Function;
+
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.RouteDefinition;
+import org.apache.karaf.camel.itests.AbstractCamelSingleFeatureResultMockBasedRouteSupplier;
+import org.osgi.service.component.annotations.Component;
+
+
```
Comment (essobedo) [inline tests/features/camel-atom/src/main/java/org/apache/karaf/camel/test/CamelAtomRouteSupplier.java:25]:
```suggestion
```

File: tests/features/camel-atom/src/test/java/org/apache/karaf/camel/itest/CamelAtomITest.java (hunk 3 above)
Comment (f2par0) [reply tests/features/camel-atom/src/test/java/org/apache/karaf/camel/itest/CamelAtomITest.java]: no, fixed

**Review by essobedo — APPROVED** [review]
LGTM

**Pull request merged**